### PR TITLE
[Merged by Bors] - refactor(Topology): simplify proofs that `CompHaus` and friends are `Precoherent`

### PR DESCRIPTION
**Pull request by dagurtomas**
We prove that `CompHaus` and friends are `Precoherent` by proving that they are `Preregular` and `FinitaryExtensive`, removing the previous hacky proofs of characterisations of effective epimorphic families. We also give new, cleaner proofs of those characterisations.

- [x] depends on: #8643 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Category/CompHaus/EffectiveEpi.lean [hunk 1]
```diff
@@ -223,38 +129,13 @@ theorem effectiveEpiFamily_tfae
     simp only [Discrete.functor_obj, colimit.ι_desc, Cofan.mk_pt, Cofan.mk_ι_app,
       colimit.comp_coconePointUniqueUpToIso_hom_assoc]
     ext; rfl
-  tfae_have 3 → 1
-  · apply effectiveEpiFamily_of_jointly_surjective
   tfae_finish
 
-instance precoherent : Precoherent CompHaus.{u} := by
```
Comment (jcommelin) [inline Mathlib/Topology/Category/CompHaus/EffectiveEpi.lean:230]: The instance for `Preregular` moved up, but where did this instance go?

**Review by jcommelin — COMMENTED** [review]
It seems to me that all the `Precoherent` instances somehow got dropped. Do I misunderstand what is happening?

File: Mathlib/Topology/Category/Profinite/EffectiveEpi.lean [hunk 2]
```diff
@@ -7,304 +7,133 @@ import Mathlib.CategoryTheory.Sites.RegularExtensive
 import Mathlib.Topology.Category.Profinite.Limits
 
 /-!
-# Effective epimorphic families in `Profinite`
+# Effective epimorhpisms and finite effective epimorphic families in `Profinite`
```
Comment (jcommelin) [inline Mathlib/Topology/Category/Profinite/EffectiveEpi.lean:10]:
```suggestion
# Effective epimorphisms and finite effective epimorphic families in `Profinite`
```

File: Mathlib/Topology/Category/CompHaus/EffectiveEpi.lean (hunk 1 above)
Comment (dagurtomas) [reply Mathlib/Topology/Category/CompHaus/EffectiveEpi.lean:230]: `Precoherent` now follows from `Preregular` and `FinitaryPreExtensive`. So the proof would be `inferInstance`. Then I'm not supposed to state it, right? Or maybe it should be stated as an `example`?

Comment (jcommelin) [reply Mathlib/Topology/Category/CompHaus/EffectiveEpi.lean:230]: I think stating it as an example is helpful. To remind people like me of this implication (-;

**Pull request closed without merging**